test/unit/math/prim/fun/value_of_test.cpp

```diff
@@ -40,30 +40,6 @@ TEST(MathMatrixPrimArr, value_of) {
     EXPECT_FLOAT_EQ(a[i], d_a[i]);
 }
 
-TEST(MathFunctions, value_of_int_return_type_short_circuit) {
-  std::vector<int> a(5, 0);
-  EXPECT_FALSE((std::is_same<decltype(stan::math::value_of(a)),
-                             std::vector<int>>::value));
-  EXPECT_FALSE((std::is_same<decltype(stan::math::value_of(a)),
-                             std::vector<int>&>::value));
-  EXPECT_FALSE((std::is_same<decltype(stan::math::value_of(a)),
-                             const std::vector<int>>::value));
-  EXPECT_TRUE((std::is_same<decltype(stan::math::value_of(a)),
-                            const std::vector<int>&>::value));
-}
-
-TEST(MathFunctions, value_of_double_return_type_short_circuit) {
-  std::vector<double> a(5, 0);
-  EXPECT_FALSE((std::is_same<decltype(stan::math::value_of(a)),
-                             std::vector<double>>::value));
-  EXPECT_FALSE((std::is_same<decltype(stan::math::value_of(a)),
-                             std::vector<double>&>::value));
-  EXPECT_FALSE((std::is_same<decltype(stan::math::value_of(a)),
-                             const std::vector<double>>::value));
-  EXPECT_TRUE((std::is_same<decltype(stan::math::value_of(a)),
-                            const std::vector<double>&>::value));
-}
-
```

Comment:
You could modify these tests so that they work with new code instead of deleting them. You can check what I did in #1828, where I did similar changes to `value_of`.